### DISCUSSION TITLE
[Question] Minirocket input dimensions?

### DISCUSSION THREAD

**Discussion by cheulyop** [timeseriesAI/tsai#149] [Q&A] [answered]
I'm trying to use Minirocket for ts classification and I'm wondering do input dimensions don't matter for Minirocket?

Typically, we'd have an input of shape [N, L, C] if we were to train a PyTorch RNN.

Likewise, don't Minirocket need its inputs to be shaped a certain way?

I couldn't find any information on this neither on the documentation and the source code.

Also, what does each dimension of sample data mean? For example:
```
dsid = 'NATOPS'
X_train, y_train, X_test, y_test = get_UCR_data(dsid)
```
What are the resulting dimensions of X_train?

**Comment by oguiza** [comment, marked as answer]
Hi @cheulyop,
I would recommend you to take a look at the [MiniRocket tutorial notebook](https://github.com/timeseriesAI/tsai/blob/main/tutorial_nbs/10_Time_Series_Classification_and_Regression_with_MiniRocket.ipynb) since there are different versions of MiniRocket. I'm not sure to which one you are referring to.

**Reply by cheulyop** [reply]
@oguiza Sure, maybe I wasn't clear enough. I did go through the notebook and had a MiniRocket model working before asking this question. However, the notebook didn't clear my question since I couldn't find in the notebook (or in the documentation) the shape of X_train.

My question applies to both the sklearn-type and the PyTorch implementation. What should be the shape of `X_train` which `model.fit(X_train, y_train)` takes as an input? Or I suppose it's unlikely, but does it not matter?
```
# Univariate classification with sklearn-type API
dsid = 'OliveOil'
X_train, y_train, X_valid, y_valid = get_UCR_data(dsid)   # Download the UCR dataset

# Computes MiniRocket features using the original (non-PyTorch) MiniRocket code.
# It then sends them to a sklearn's RidgeClassifier (linear classifier).
model = MiniRocketClassifier()
timer.start(False)
model.fit(X_train, y_train)  # <- what's the shape of X_train here?
t = timer.stop()
print(f'valid accuracy    : {model.score(X_valid, y_valid):.3%} time: {t}')
```
```
# pytorch code
mrf = MiniRocketFeatures(X.shape[1], X.shape[2]).to(default_device())
X_train = torch.from_numpy(X[splits[0]]).to(default_device())
mrf.fit(X_train)  # <- same here, what should be the shape of X_train?
X_feat = get_minirocket_features(X, mrf, chunksize=1024, to_np=True)
X_feat.shape, type(X_feat)
```

Thanks,
Cheul

**Reply by oguiza** [reply]
X_train in MiniRocketClassifier/ MiniRocketRegressor and MiniRocketFeatures must have a shape [n_samples x n_variables x sequence length], same as any other model in `tsai`.
MiniRocket will then apply approx 10k convolutions and calculate a single feature per convolution applied. So the output has the same shape, independently of the input.

**Reply by cheulyop** [reply]
Amazing, thank you for the clarification.
